tests/test_read_mri_data.py

```diff
@@ -0,0 +1,82 @@
+from mpi4py import MPI
+import dolfinx
+import numpy as np
+from scifem.biomedical import read_mri_data_to_function, read_mri_data_to_tag
+import pytest
+import basix.ufl
+
+
+@pytest.mark.parametrize("degree", [0, 1, 3])
+@pytest.mark.parametrize("M", [25, 17])
+@pytest.mark.parametrize("Nx", [7, 3])
+@pytest.mark.parametrize("Ny", [23, 8])
+@pytest.mark.parametrize("Nz", [9, 17])
+@pytest.mark.parametrize("theta", [np.pi / 3, 0, -np.pi])
+@pytest.mark.parametrize("translation", [np.array([0, 0, 0]), np.array([2.1, 1.3, 0.4])])
+def test_read_mri_data_to_function(degree, M, Nx, Ny, Nz, theta, translation, tmp_path):
+    nibabel = pytest.importorskip("nibabel")
+
+    # Generate rotation and scaling matrix equivalent to a unit cube
+    rotation_matrix_3D = np.array(
+        [[1, 0, 0], [0, np.cos(theta), -np.sin(theta)], [0, np.sin(theta), np.cos(theta)]]
+    )
+    scale_matrix = 1.0 / M * np.identity(3)
+
+    # Generate the affien mapping for nibabel
+    A = np.append(np.dot(rotation_matrix_3D, scale_matrix), (translation).reshape(3, 1), axis=1)
+    A = np.vstack([A, [0, 0, 0, 1]])
+
+    # Write transformed data to file
+    data = np.arange(1, M**3 + 1, dtype=np.float64).reshape(M, M, M)
+    image = nibabel.nifti1.Nifti1Image(data, affine=A)
+
+    path = MPI.COMM_WORLD.bcast(tmp_path, root=0)
+    filename = path.with_suffix(".mgz")
+
+    if MPI.COMM_WORLD.rank == 0:
+        nibabel.save(image, filename)
+    MPI.COMM_WORLD.Barrier()
+    # Create unit cube
+    mesh = dolfinx.mesh.create_unit_cube(
+        MPI.COMM_WORLD,
+        Nx,
+        Ny,
+        Nz,
+        cell_type=dolfinx.cpp.mesh.CellType.hexahedron,
+        ghost_mode=dolfinx.mesh.GhostMode.shared_facet,
+    )
+
+    # Find voxel position on reference cube (done prior to mesh transformation)
+    q_el = basix.ufl.quadrature_element(mesh.topology.cell_name(), degree=degree)
+    V = dolfinx.fem.functionspace(mesh, q_el)
+    dx = 1 / M
+    coords = np.abs(V.tabulate_dof_coordinates())  # Remove rounding error
```

Comment:
Why `np.abs` here? And what does the comment "Remove rounding error" refers to here?

Reply:
Tabulate dof coordinates at the origin of a unit square can be -1e-16, meaning that the computed voxel gets rounded to -1.